python/ray/function_manager.py

```diff
@@ -556,16 +604,64 @@ def export_actor_class(self, Class, actor_method_names,
             # because of https://github.com/ray-project/ray/issues/1146.
 
     def load_actor(self, driver_id, function_descriptor):
-        key = (b"ActorClass:" + driver_id + b":" +
-               function_descriptor.function_id.id())
-        # Wait for the actor class key to have been imported by the
-        # import thread. TODO(rkn): It shouldn't be possible to end
-        # up in an infinite loop here, but we should push an error to
-        # the driver if too much time is spent here.
-        while key not in self.imported_actor_classes:
-            time.sleep(0.001)
-        with self._worker.lock:
-            self.fetch_and_register_actor(key)
+        function_id_bytes = function_descriptor.function_id.id()
+        actor_class = self._loaded_actor_classes.get(function_id_bytes, None)
+        if actor_class is None:
+            if self._worker.load_code_from_local:
+                try:
+                    cls = self._load_actor_from_local(driver_id,
+                                                      function_descriptor)
+                    self._loaded_actor_classes[function_id_bytes] = cls
+                    self._worker.actor_class = cls
+                    self._worker.actor_checkpoint_interval = 0
+                    actor_id_str = self._worker.actor_id
+                    self._worker.actors[actor_id_str] = cls.__new__(cls)
+                    return
+                except AttributeError as e:
+                    logger.warning("Cannot find function {} in local files,"
+                                   " with message: {} turn to GCS".format(
+                                       function_descriptor.function_name, e))
+
+            key = (b"ActorClass:" + driver_id + b":" + function_id_bytes)
+            # Wait for the actor class key to have been imported by the
+            # import thread. TODO(rkn): It shouldn't be possible to end
+            # up in an infinite loop here, but we should push an error to
+            # the driver if too much time is spent here.
+            while key not in self.imported_actor_classes:
+                time.sleep(0.001)
+            with self._worker.lock:
+                self.fetch_and_register_actor(key)
+        else:
+            self._worker.actor_class = cls
+            self._worker.actor_checkpoint_interval = 0
+            actor_id_str = self._worker.actor_id
+            self._worker.actors[actor_id_str] = cls.__new__(cls)
+
+    def _load_actor_from_local(self, driver_id, function_descriptor):
```

Comment:
This function has duplicated code with `fetch_and_register_actor`. We can combine them

Reply:
Also, I think we can get rid of the `load_actor` method. When the task is an actor creation task, `function_actor_manager.get_execution_info` just returns the constructor of the actor.

Reply:
Thanks for your suggestions. I used a function to remove the duplicated code. For the second suggestion, I think it is better to implement it in another PR.